company-lsp.el

```diff
@@ -305,22 +328,15 @@ See the documentation of `company-backends' for COMMAND and ARG."
              (lsp--capability "completionProvider")
              (or (company-lsp--completion-prefix) 'stop)))
     (candidates
-     (cons :async
-           #'(lambda (callback)
-               (let ((cache (company-lsp--cache-get arg)))
-                 (if cache
-                     (funcall callback (plist-get cache :candidates))
-                   (let ((req (lsp--make-request "textDocument/completion"
-                                                 (lsp--text-document-position-params))))
-                     (setq company-lsp--line-backup
-                           (buffer-substring (line-beginning-position) (line-end-position)))
-                     (if company-lsp-async
-                         (lsp--send-request-async req
-                                                  (lambda (resp)
-                                                    (company-lsp--on-completion resp arg callback)))
-                       (company-lsp--on-completion (lsp--send-request req)
-                                                   arg
-                                                   callback))))))))
+     (setq company-lsp--line-backup
```

Comment:
I'm not sure why you need to save this, TBH.

Reply:
That's a sign for documentation. Added.